chia/full_node/coin_store.py

```diff
@@ -409,13 +427,27 @@ async def _add_coin_records(self, records: List[CoinRecord]) -> None:
                     record.coin.parent_coin_info.hex(),
                     bytes(record.coin.amount),
                     record.timestamp,
+                    txId,
                 )
             )
 
         cursor = await self.coin_record_db.executemany(
-            "INSERT INTO coin_record VALUES(?, ?, ?, ?, ?, ?, ?, ?, ?)",
+            "INSERT INTO temp_coin_record VALUES(?, ?, ?, ?, ?, ?, ?, ?, ?, ?)",
             values,
         )
+
+        cursor = await self.coin_record_db.execute(
+            "INSERT INTO coin_record (coin_name, confirmed_index, spent_index, spent, coinbase, puzzle_hash, coin_parent, amount, timestamp)"
+            "     SELECT coin_name, confirmed_index, spent_index, spent, coinbase, puzzle_hash, coin_parent, amount, timestamp"
+            "       FROM temp_coin_record"
+            "      WHERE tx_id = ?",
+            (txId,),
+        )
+
+        await self.coin_record_db.execute(
+            "DELETE FROM temp_coin_record WHERE tx_id = ?", (txId,),
```

Comment:
actually, I'm pretty sure we won't ever interleave calls to this function, so we know we need to drop every row in the temp table here.

Reply:
So you think, we don't even need a transaction id? That would make it even more simple.

Reply:
I'm pretty confident calls to this function are not interleaved